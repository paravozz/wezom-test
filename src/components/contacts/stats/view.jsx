import React from 'react';
import { Typography, Col, Row, Statistic, Descriptions } from 'antd';
import {
	countGenders,
	getGenderPredominateMessage,
	countNationalities,
	CountryData,
} from 'utils';


const View = (props) => {
	const { contacts, loading } = props;

	if (!contacts?.length || loading) {
		return (
			<Row className="box" />
		);
	}

	function renderNationalitiesStats(_contacts) {
		const natCount = countNationalities(_contacts);

		return (
			<Descriptions title="Nationalities">
				{Object.entries(natCount).map(([key, value]) => (
					<Descriptions.Item key={key} label={CountryData[key].name}>
						{value} contacts
					</Descriptions.Item>
				))}
			</Descriptions>
		);
	}

	const genderStats = countGenders(contacts);
	const genderPredominateMessage = getGenderPredominateMessage(genderStats);

	return (
		<div className="box">
			<Typography.Title level={3}>Statistic</Typography.Title>
			<Row gutter={24}>
				<Col className="_flex-noshrink">
					<Statistic title="Collection size" value={contacts.length} />
				</Col>
				<Col className="_flex-noshrink">
					<Row gutter={12}>
						<Col span={6}>
							<Statistic title="Males" value={genderStats.male} />
						</Col>
						<Col span={6}>
							<Statistic title="Females" value={genderStats.female} />
						</Col>
						<Col span={6}>
							<Statistic title="Indeterminate" value={genderStats.indeterminate} />
						</Col>
						<Col span={24}>
							<Typography.Text mark>
								{genderPredominateMessage}
							</Typography.Text>
						</Col>
					</Row>
				</Col>
			</Row>
			<br />
			<Row>
				{renderNationalitiesStats(contacts)}
			</Row>
		</div>
	);
};


export { View };
