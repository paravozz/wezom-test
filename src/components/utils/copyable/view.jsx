import React from 'react';

import { CopyToClipboard } from 'react-copy-to-clipboard';
import { CopyOutlined, CheckOutlined } from '@ant-design/icons';
import { Tooltip } from 'antd';

import './style.scss';


export default ({ children, text }) => {
	const [isCopied, setIsCopied] = React.useState(false);


	React.useEffect(() => {
		if (isCopied) {
			setTimeout(() => {
				setIsCopied(false);
			}, 3000);
		}
	}, [isCopied]);


	return (
		<div className="copyable--container">
			<CopyToClipboard onCopy={() => setIsCopied(true)} text={text}>
				<Tooltip title={isCopied ? 'Copied' : 'Copy'}>
					{
						isCopied
							? <CheckOutlined style={{ color: '#52c41a' }} />
							: <CopyOutlined style={{ color: '#1890ff' }} />
					}
				</Tooltip>
			</CopyToClipboard>
			{children}
		</div>
	)
};
