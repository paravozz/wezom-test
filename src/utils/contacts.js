export function getFullName(title, first, last) {
	return `${title} ${first} ${last}`;
}

const mapGenderToItsName = {
	male: 'Men',
	female: 'Women',
	indeterminate: 'Indeterminate',
};

export function getGenderName(gender) {
	if (!gender) {
		return mapGenderToItsName.indeterminate;
	}

	return mapGenderToItsName[gender];
}

export function paginateCollection(collection, offset = 0, limit = 10) {
	return collection.slice(offset, offset + limit);
}
