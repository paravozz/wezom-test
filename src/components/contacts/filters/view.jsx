import React from 'react';
import { Row, Input, Select, Col, Checkbox, Button } from 'antd';
import { CloseOutlined } from '@ant-design/icons';
import { CountryData } from 'utils';
import { getGenderName } from 'utils';


const View = (props) => {
	const {
		loading,
		filters,
		filterContacts,
	} = props;

	return (
		<div className="box">
			<Row align="middle">
				<Col className="_flex-grow">
					<Row gutter={[6, 6]} align="middle">
						<Col xs={24} lg={11}>
							<Row>
								<Input.Search
									loading={loading}
									placeholder="Search by full name"
									size="large"
									onChange={e => filterContacts({ ...filters, name: e.target.value })}
									value={filters.name}
								/>
							</Row>
						</Col>
						<Col xs={24} sm={10} lg={4}>
							<Row>
								<Select
									id="genderSelect"
									loading={loading}
									style={{ width: '100%' }}
									allowClear
									placeholder="Gender"
									size="large"
									onChange={value => filterContacts({ ...filters, gender: value })}
									value={filters.gender || undefined}
								>
									{['male', 'female', 'indeterminate'].map(k => (
										<Select.Option key={k} value={k}>
											{getGenderName(k)}
										</Select.Option>
									))}
								</Select>
							</Row>
						</Col>
						<Col xs={24} sm={14} lg={5}>
							<Row>
								<Select
									id="natSelect"
									loading={loading}
									style={{ width: '100%' }}
									mode="multiple"
									maxTagCount={2}
									allowClear
									size="large"
									placeholder="Nationality"
									onChange={value => filterContacts({ ...filters, nat: value })}
									value={filters.nat}
								>
									{Object.entries(CountryData).map(([key, data]) => (
										<Select.Option key={key} value={key}>{data.name}</Select.Option>
									))}
								</Select>
							</Row>
						</Col>
						<Col xs={24} sm={6} lg={4}>
							<Row>
								<Checkbox
									disabled={loading}
									onChange={(e) => filterContacts({ ...filters, author: e.target.checked })}
								>
									I am creator
								</Checkbox>
							</Row>
						</Col>
					</Row>
				</Col>
				<Col className="_flex-noshrink">
					<Row>
						<Button
							disabled={loading}
							onClick={() => filterContacts(null)}
							type="link"
						>
							<CloseOutlined /> Clear
						</Button>
					</Row>
				</Col>
			</Row>
		</div>
	);
};


export { View };
