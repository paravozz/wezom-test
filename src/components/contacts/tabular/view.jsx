import React from 'react';
import { Table, Typography, Tag, Avatar, Empty } from 'antd';
import { UserOutlined } from '@ant-design/icons';
import moment from 'moment';
import { CountryData, getFullName } from 'utils';
import { Copyable } from 'components/utils';


const View = (props) => {
	const { contacts, loading } = props;

	if (!contacts) {
		return (
			<>
				<Empty image={Empty.PRESENTED_IMAGE_SIMPLE} />
			</>
		)
	}

	function renderFullName({ title, first, last }) {
		return getFullName(title, first, last);
	}

	function renderAvatar(pictureUrl) {
		return (
			<Avatar size={40} icon={<UserOutlined />} src={pictureUrl} />
		);
	}

	function renderBirthday({ date, age }) {
		return (
			<div>
				{moment(date).format('dddd, M/D/YYYY, H:m:s A')}
				<br />
				{age} years
			</div>
		);
	}

	function renderEmail(email) {
		return (
			<Copyable text={email}>
				<Typography.Link ellipsis href={`mailto:${email}`}>
					{email}
				</Typography.Link>
			</Copyable>
		);
	}

	function renderPhone(phone) {
		return (
			<Copyable text={phone}>
				<Typography.Link ellipsis href={`tel:${phone}`}>
					{phone}
				</Typography.Link>
			</Copyable>
		);
	}

	function renderLocation({ country, street, city, state, postcode }) {
		return (
			<Copyable
				text={(
					`[${country}] `
					+ `${street.number} ${street.name}, `
					+ `${city}, ${state} ${postcode}`
				)}
			>
				<Typography.Paragraph ellipsis={{ rows: 3 }}>
					<strong>/{country}/</strong>
					<br />
					{(
						`${street.number}, ${street.name}`
						+ `${city}, ${state} ${postcode}`
					)}
				</Typography.Paragraph>
			</Copyable>
		);
	}

	function renderNationality(nat) {
		return (
			<Tag color={CountryData[nat].color}>{CountryData[nat].name}</Tag>
		)
	}

	const dataSource = (
		contacts
			.map((contact, index) => ({
				key: index,
				avatar: renderAvatar(contact.picture.thumbnail),
				fullName: renderFullName(contact.name),
				birthday: renderBirthday(contact.dob),
				email: renderEmail(contact.email),
				phone: renderPhone(contact.phone),
				location: renderLocation(contact.location),
				nationality: renderNationality(contact.nat),
			}))
	);

	const columns = [
		{
			title: 'Avatar',
			dataIndex: 'avatar',
			key: 'avatar',
			fixed: 'left',
			width: 80,
		},
		{
			title: 'Full name',
			dataIndex: 'fullName',
			key: 'fullName',
			width: 155,
			sorter: (a, b) => a.fullName.length - b.fullName.length,
			sortDirections: ['descend', 'ascend'],
		},
		{
			title: 'Birthday',
			dataIndex: 'birthday',
			key: 'birthday',
			width: 250,
		},
		{
			title: 'Email',
			dataIndex: 'email',
			key: 'email',
			width: 155,
		},
		{
			title: 'Phone',
			dataIndex: 'phone',
			key: 'phone',
			width: 155,
		},
		{
			title: 'Location',
			dataIndex: 'location',
			key: 'location',
			width: 250,
		},
		{
			title: 'Nationality',
			dataIndex: 'nationality',
			key: 'nationality',
			width: 155,
			align: 'right',
		},
	];


	return (
		<Table
			dataSource={dataSource}
			columns={columns}
			scroll={{ x: '100vw' }}
			loading={loading}
			pagination={{ position: ['bottomCenter'] }}
		/>
	);
};


export { View };
