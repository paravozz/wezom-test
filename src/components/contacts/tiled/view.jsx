import React from 'react';
import { Typography, Tag, Row, Col, Card, Divider, Empty, Spin } from 'antd';
import { CountryData, getFullName, paginateCollection } from 'utils';
import { Copyable } from 'components/utils';


function renderFullName({ title, first, last }) {
	return getFullName(title, first, last);
}

function renderAvatar(pictureUrl) {
	return (
		<img src={pictureUrl} alt="contact-cover" />
	);
}

function renderBirthday({ age }) {
	return `${age} years`
}

function renderEmail(email) {
	return (
		<Copyable text={email}>
			<Typography.Link ellipsis href={`mailto:${email}`}>
				{email}
			</Typography.Link>
		</Copyable>
	);
}

function renderPhone(phone) {
	return (
		<Copyable text={phone}>
			<Typography.Link ellipsis href={`tel:${phone}`}>
				{phone}
			</Typography.Link>
		</Copyable>
	);
}

function renderLocation({ country, street, city, state, postcode }) {
	return (
		<Copyable
			text={(
				`[${country}] `
				+ `${street.number} ${street.name}, `
				+ `${city}, ${state} ${postcode}`
			)}
		>
			<Typography.Paragraph ellipsis={{ rows: 3 }}>
				<strong>/{country}/</strong>
				<br />
				{(
					`${street.number}, ${street.name}`
					+ `${city}, ${state} ${postcode}`
				)}
			</Typography.Paragraph>
		</Copyable>
	);
}

function renderNationality(nat) {
	return (
		<Tag color={CountryData[nat].color}>{CountryData[nat].name}</Tag>
	)
}

export function transformContacts(pagination, contacts) {
	let baseCollection = (
		pagination
			? paginateCollection(contacts, pagination.offset, pagination.limit)
			: contacts
	);

	return baseCollection.map((contact, index) => ({
		key: index,
		avatar: renderAvatar(contact.picture.large),
		fullName: renderFullName(contact.name),
		birthday: renderBirthday(contact.dob),
		email: renderEmail(contact.email),
		phone: renderPhone(contact.phone),
		location: renderLocation(contact.location),
		nationality: renderNationality(contact.nat),
	}));
}


const View = (props) => {
	const { contacts, loading, pagination } = props;

	if (!contacts?.length || loading) {
		return (
			<>
				{loading && <Spin />}
				<Empty image={Empty.PRESENTED_IMAGE_SIMPLE} />
			</>
		)
	}

	return (
		<Row gutter={[8, 8]}>
			{transformContacts(pagination, contacts).map(contact => (
				<div key={contact.key} style={{ width: '33.3333%', maxWidth: '33.3333%' }}>
					<Col>
						<Card
							hoverable
							cover={contact.avatar}
						>
							<Card.Meta
								title={(
									<Typography.Text>
										<Typography.Text>{contact.fullName}</Typography.Text>
										&nbsp;
										<Typography.Text type="secondary">
											({contact.birthday})
										</Typography.Text>
									</Typography.Text>
								)}
								description={(
									<>
										{contact.email}
										{contact.phone}
										{contact.location}
										<Divider/>
										{contact.nationality}
									</>
								)}
							/>
						</Card>
					</Col>
				</div>
			))}
		</Row>
	);
};


export { View };
