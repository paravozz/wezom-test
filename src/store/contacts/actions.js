import { createAction } from '../utils';


export const OActionTypes = {
	fetchContactsRequested: 'CONTACTS_FETCH_REQUESTED',
	fetchContactsSucceed: 'CONTACTS_FETCH_SUCCEEDED',
	fetchContactsFailed: 'CONTACTS_FETCH_FAILED',

	filtersChangeRequested: 'CONTACTS_FILTERS_CHANGE_REQUESTED',
	filtersChanged: 'CONTACTS_FILTERS_CHANGED',
	filtersReset: 'CONTACTS_FILTERS_RESET',

	paginationRequested: 'CONTACTS_PAGINATION_REQUESTED',
	paginationSet: 'CONTACTS_PAGINATION_SET',
};


export const fetchContacts = (count, seed) => (
	createAction(OActionTypes.fetchContactsRequested)({ count, seed })
);

export const filterContacts = (filters) => (
	createAction(OActionTypes.filtersChangeRequested)({ filters })
);

export const setPagination = (pagination) => (
	createAction(OActionTypes.paginationRequested)({ pagination })
);
