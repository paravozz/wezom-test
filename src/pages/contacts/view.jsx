import React from 'react';

import { Radio, Row, Col, Typography, Pagination, Tooltip, Button } from 'antd';
import { DatabaseOutlined, AppstoreOutlined, ReloadOutlined } from '@ant-design/icons';

import { getFromStorage, randomIntegerInRange, setToStorage } from 'utils';
import {
	TabularContacts,
	StatsForContacts,
	FiltersForContacts,
} from 'components';
import { TiledContacts } from '../../components/contacts/tiled';


const __VIEW_MODE_KEY__ = 'app__view_mode';


const View = (props) => {
	const {
		totalCount,
		fetchContacts,
		pagination,
		setPagination,
		contacts,
	} = props;

	const [viewMode, setViewMode] = React.useState(getFromStorage(__VIEW_MODE_KEY__) || 'tabular');

	function fetchContactsRemote() {
		fetchContacts(totalCount || randomIntegerInRange(1, 1000));
	}


	React.useEffect(() => {
		fetchContactsRemote();
		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, []);

	React.useEffect(() => {
		setToStorage(__VIEW_MODE_KEY__, viewMode);
	}, [viewMode]);


	const radioOptions = [
		{ label: <DatabaseOutlined />, value: 'tabular' },
		{ label: <AppstoreOutlined />, value: 'tiled' },
	];

	const componentForViewMode = {
		tiled: (
			<>
				<TiledContacts />
				<Pagination
					onChange={(page, pageSize) => (
						setPagination({
							offset: pageSize * (page - 1),
							limit: pageSize,
						})
					)}
					current={pagination.offset / pagination.limit + 1}
					pageSize={pagination.limit}
					pageSizeOptions={[6, 12, 24, 36, 48]}
					total={contacts.length}
				/>
			</>
		),
		tabular: <TabularContacts />,
	};


	return (
		<div className={'page page--contacts'}>
			<Row justify="space-between">
				<Typography.Title level={2}>Contacts</Typography.Title>
				<Col>
					<Tooltip title="Reload data">
				  		<Button
							type="dashed"
							shape="circle"
							icon={<ReloadOutlined />}
							onClick={() => fetchContactsRemote()}
						/>
					</Tooltip>
					<Radio.Group
						options={radioOptions}
						onChange={(e) => setViewMode(e.target.value)}
						value={viewMode}
						optionType="button"
						buttonStyle="solid"
					/>
				</Col>
			</Row>
			<FiltersForContacts />
			{componentForViewMode[viewMode]}
			<StatsForContacts />
		</div>
	);
};


export { View };
