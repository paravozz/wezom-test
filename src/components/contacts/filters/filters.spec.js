/* eslint-disable */
import React from 'react';
import { View as Filters } from './view.jsx';
import { Input, Select } from 'antd';

import * as actions from 'store/contacts/actions';


const setUp = (props) => shallow(<Filters {...props} />);
const setUpRendered = (props) => render(<Filters {...props} />);

describe('Filters', () => {
	let component;

	const defaultFilters = {
		name: '',
		gender: '',
		nat: [],
		author: '',
	};

	beforeEach(() => {
		jest.resetModules();
		component = setUp({ filters: defaultFilters });
	});

	it('should render without crashing', () => {
		component = setUpRendered({ filters: defaultFilters });
		expect(component).toMatchSnapshot();
	});

	it('should render box', () => {
		expect((component.find('.box')?.length)).toBe(1);
	});

	test('filters should filter as expected', () => {
		const filterContactsMock = jest.fn((...args) => {});
		component = setUp({ filters: defaultFilters, filterContacts: filterContactsMock });

		const searchComponent = component.find(Input.Search);
		const selectGenderComponent = component.findWhere((el) => el.props().id === 'genderSelect');
		const selectNatComponent = component.findWhere((el) => el.props().id === 'natSelect');

		expect(searchComponent.props().value).toBe(defaultFilters.name);
		expect(selectGenderComponent.props().value).toBe(defaultFilters.gender || undefined);
		expect(selectNatComponent.props().value).toBe(defaultFilters.nat);

		searchComponent.simulate('change', { target: { value: 'test-search' } });
		expect(filterContactsMock).toBeCalledWith({ ...defaultFilters, name: 'test-search' });
		filterContactsMock.mockReset();

		selectGenderComponent.simulate('change', 'male');
		expect(filterContactsMock).toBeCalledWith({ ...defaultFilters, gender: 'male' });
		filterContactsMock.mockReset();

		selectNatComponent.simulate('change', 'RU');
		expect(filterContactsMock).toBeCalledWith({ ...defaultFilters, nat: 'RU' });
		filterContactsMock.mockReset();

		filterContactsMock.mockClear();
	})
});
