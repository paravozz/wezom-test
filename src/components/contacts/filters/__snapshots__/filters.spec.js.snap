// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters should render without crashing 1`] = `
<div
  class="box"
>
  <div
    class="ant-row ant-row-middle"
  >
    <div
      class="ant-col _flex-grow"
    >
      <div
        class="ant-row ant-row-middle"
        style="margin-left:-3px;margin-right:-3px;margin-top:-3px;margin-bottom:3px"
      >
        <div
          class="ant-col ant-col-xs-24 ant-col-lg-11"
          style="padding-left:3px;padding-right:3px;padding-top:3px;padding-bottom:3px"
        >
          <div
            class="ant-row"
          >
            <span
              class="ant-input-search ant-input-affix-wrapper ant-input-affix-wrapper-lg"
            >
              <input
                class="ant-input ant-input-lg"
                placeholder="Search by full name"
                type="text"
                value=""
              />
              <span
                class="ant-input-suffix"
              >
                <span
                  aria-label="search"
                  class="anticon anticon-search ant-input-search-icon"
                  role="img"
                  tabindex="-1"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="search"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                    />
                  </svg>
                </span>
              </span>
            </span>
          </div>
        </div>
        <div
          class="ant-col ant-col-xs-24 ant-col-sm-10 ant-col-lg-4"
          style="padding-left:3px;padding-right:3px;padding-top:3px;padding-bottom:3px"
        >
          <div
            class="ant-row"
          >
            <div
              class="ant-select ant-select-lg ant-select-single ant-select-allow-clear ant-select-show-arrow"
              style="width:100%"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="genderSelect_list_0"
                    aria-autocomplete="list"
                    aria-controls="genderSelect_list"
                    aria-haspopup="listbox"
                    aria-owns="genderSelect_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="genderSelect"
                    readonly=""
                    role="combobox"
                    style="opacity:0"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-placeholder"
                >
                  Gender
                </span>
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select:none;-webkit-user-select:none"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-select-suffix"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-col ant-col-xs-24 ant-col-sm-14 ant-col-lg-5"
          style="padding-left:3px;padding-right:3px;padding-top:3px;padding-bottom:3px"
        >
          <div
            class="ant-row"
          >
            <div
              class="ant-select ant-select-lg ant-select-multiple ant-select-allow-clear ant-select-show-search"
              style="width:100%"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                  style="width:0"
                >
                  <input
                    aria-activedescendant="natSelect_list_0"
                    aria-autocomplete="list"
                    aria-controls="natSelect_list"
                    aria-haspopup="listbox"
                    aria-owns="natSelect_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="natSelect"
                    readonly=""
                    role="combobox"
                    style="opacity:0"
                    unselectable="on"
                    value=""
                  />
                  <span
                    aria-hidden="true"
                    class="ant-select-selection-search-mirror"
                  >
                    Â 
                  </span>
                </span>
                <span
                  class="ant-select-selection-placeholder"
                >
                  Nationality
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-col ant-col-xs-24 ant-col-sm-6 ant-col-lg-4"
          style="padding-left:3px;padding-right:3px;padding-top:3px;padding-bottom:3px"
        >
          <div
            class="ant-row"
          >
            <label
              class="ant-checkbox-wrapper"
            >
              <span
                class="ant-checkbox"
              >
                <input
                  class="ant-checkbox-input"
                  type="checkbox"
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span>
                I am creator
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col _flex-noshrink"
    >
      <div
        class="ant-row"
      >
        <button
          class="ant-btn ant-btn-link"
          type="button"
        >
          <span
            aria-label="close"
            class="anticon anticon-close"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </span>
          <span>
             Clear
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
